### PR TITLE
Ensure config module loads opts from the named_context.

### LINKED ISSUE #61297: deltaproxy overriding minion configuration settings
### Description of Issue
I am evaluating the deltaproxy feature released on 3004 version and noticed the configuration settings for other deltaproxy managed minions are getting overridden with the settings defined for the last minion defined under "ids". This results in cases like targeting one minion to execute a command and getting the result from another one. This behaviour was originally observed for napalm and netmiko proxies and, to make sure it's nothing related to my environment, I brought up a fresh 3004 instance with 1 deltaproxy and 3 dummy proxies. Details below:

### Setup
_/etc/salt/master_:
```
interface: 127.0.0.1
```

_/etc/salt/proxy_:
```
master: 127.0.0.1
metaproxy: deltaproxy
enable_fqdns_grains: False
multiprocessing: True
```

_/srv/pillar/top.sls_:
```
base:
  deltaproxy:
    - deltaproxy
  dummy1:
    - dummy1
  dummy2:
   - dummy2
  dummy3:
   - dummy3
```

_/srv/pillar/deltaproxy.sls_:
```
proxy:
  proxytype: deltaproxy
  ids:
    - dummy1
    - dummy2
    - dummy3
```

_/srv/pilar/dummy1.sls_:
```
proxy:
  proxytype: dummy
  hostname: dummy1
```

_/srv/pilar/dummy2.sls_:
```
proxy:
  proxytype: dummy
  hostname: dummy2
```

_/srv/pilar/dummy3.sls_:
```
proxy:
  proxytype: dummy
  hostname: dummy3
```

### Steps to Reproduce Issue
Here it's possible to observe that _id_ and also settings defined under _pillar_ are overridden:
```
root@salt-3004:~# salt 'dummy*' config.get id
dummy1:
    dummy3
dummy2:
    dummy3
dummy3:
    dummy3

root@salt-3004:~# salt 'dummy*' config.get proxy
dummy1:
    ----------
    hostname:
        dummy3
    proxytype:
        dummy
dummy2:
    ----------
    hostname:
        dummy3
    proxytype:
        dummy
dummy3:
    ----------
    hostname:
        dummy3
    proxytype:
        dummy
```

### Versions Report
```
root@salt-3004:~# salt --versions-report
Salt Version:
          Salt: 3004

Dependency Versions:
          cffi: Not Installed
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: 4.0.5
     gitpython: 3.1.14
        Jinja2: 2.11.3
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.0
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: Not Installed
      pycrypto: Not Installed
  pycryptodome: 3.9.7
        pygit2: Not Installed
        Python: 3.9.2 (default, Feb 28 2021, 17:03:44)
  python-gnupg: Not Installed
        PyYAML: 5.3.1
         PyZMQ: 20.0.0
         smmap: 4.0.0
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.4

System Versions:
          dist: debian 11 bullseye
        locale: utf-8
       machine: x86_64
       release: 5.10.0-9-amd64
        system: Linux
       version: Debian GNU/Linux 11 bullseye
```

Let me know if there's any other information I can provide that's relevant to this issue. 
Thank you.

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Ensure opts is included in pack for minion_mods and config loads opts from the named_context.

### What issues does this PR fix or reference?
Fixes: #61297 

### Previous Behavior
When used with deltaproxy, the config module would use the `__opts__` from the last sub proxy minion that used the `config` module during initialization as part of configuring beacons. This would result in incorrect values being returned by the `config` module.

### New Behavior
Load `__opts__` from the Salt loader's named_context to ensure we're using the  `__opts__` for the correct minion.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by garethgreenaway** [comment]
run full-all

**Comment by garethgreenaway** [comment]
re-run full all

**Review by Ch3LL — CHANGES_REQUESTED** [review]
changelog

**Review by Ch3LL — DISMISSED** [review]
Approve, but want to make sure @dwoz approves too on this one.

File: tests/pytests/integration/proxy/test_deltaproxy.py [hunk 1]
```diff
@@ -137,3 +137,13 @@ def test_state_highstate(salt_master, salt_cli, tmp_path, proxy_id):
...
             assert value["result"] is True
+
+
+def test_config_get(salt_cli, proxy_id):
+    """
+    Ensure the config module returns the right id
+    when targeting deltaproxy managed proxy minions.
+    """
+    ret = salt_cli.run("config.get", "id", minion_tgt=proxy_id)
+    assert True is True
```
Comment (dwoz) [inline tests/pytests/integration/proxy/test_deltaproxy.py:148]: This looks like cruft.

**Review by dwoz — CHANGES_REQUESTED** [review]

**Review by dwoz — APPROVED** [review]

**Pull request merged**